From 629adade0f146045093eb81fdfb9f6ec8829e290 Mon Sep 17 00:00:00 2001
From: Wang Panzhenzhuan <randy.wang@rock-chips.com>
Date: Thu, 3 Dec 2020 20:28:42 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3566-rk817-tablet: fix camera ov8858
 suspend issues

cam_clkout0 correponding gpio4 A7 pinctrl may changed,
when wakedup from deep sleep. so artificial to set pinctrl.

Signed-off-by: Wang Panzhenzhuan <randy.wang@rock-chips.com>
Change-Id: Iad200cbc6685cb18217a7577746bef122fa51574
---
 .../arm64/boot/dts/rockchip/rk3566-rk817-tablet-k108.dts | 9 ++++++++-
 .../boot/dts/rockchip/rk3566-rk817-tablet-rkg11.dts      | 9 ++++++++-
 arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet.dts     | 9 ++++++++-
 3 files changed, 24 insertions(+), 3 deletions(-)
 mode change 100644 => 100755 arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet-rkg11.dts

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet-k108.dts b/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet-k108.dts
index 74075cdc5718..e116324b7de2 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet-k108.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet-k108.dts
@@ -882,8 +882,9 @@
 		clocks = <&cru CLK_CAM0_OUT>;
 		clock-names = "xvclk";
 		power-domains = <&power RK3568_PD_VI>;
-		pinctrl-names = "default";
+		pinctrl-names = "rockchip,camera_default", "rockchip,camera_sleep";
 		pinctrl-0 = <&cam_clkout0>;
+		pinctrl-1 = <&cam_sleep>;
 		//reset pin control by hardware,used this pin switch to mipi input
 		//1->2LANE(LANE 0&1) FRONT camera, 0->4LANE REAR camera
 		reset-gpios = <&gpio4 17 GPIO_ACTIVE_LOW>;
@@ -982,6 +983,12 @@
 				<4 RK_PA7 RK_FUNC_1 &pcfg_pull_none>;
 		};
 
+		cam_sleep: cam-sleep {
+			rockchip,pins =
+				/* cam_sleep */
+				<4 RK_PA7 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
 		camera_rst: camera-rst {
 			rockchip,pins =
 				/* front camera reset */
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet-rkg11.dts b/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet-rkg11.dts
old mode 100644
new mode 100755
index f35a1ed54735..3db906473d98
--- a/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet-rkg11.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet-rkg11.dts
@@ -700,8 +700,9 @@
 		clocks = <&cru CLK_CAM0_OUT>;
 		clock-names = "xvclk";
 		power-domains = <&power RK3568_PD_VI>;
-		pinctrl-names = "default";
+		pinctrl-names = "rockchip,camera_default", "rockchip,camera_sleep";
 		pinctrl-0 = <&cam_clkout0>;
+		pinctrl-1 = <&cam_sleep>;
 		//reset pin control by hardware,used this pin switch to mipi input
 		//0->FRONT camera, 1->REAR camera
 		reset-gpios = <&gpio4 17 GPIO_ACTIVE_HIGH>;
@@ -859,6 +860,12 @@
 				<4 RK_PA7 RK_FUNC_1 &pcfg_pull_none>;
 		};
 
+		cam_sleep: cam-sleep {
+			rockchip,pins =
+				/* cam_sleep */
+				<4 RK_PA7 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
 		camera_rst: camera-rst {
 			rockchip,pins =
 				/* front camera reset */
diff --git a/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet.dts b/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet.dts
index 44e09e1ab23a..330207cda330 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-rk817-tablet.dts
@@ -809,8 +809,9 @@
 		clocks = <&cru CLK_CAM0_OUT>;
 		clock-names = "xvclk";
 		power-domains = <&power RK3568_PD_VI>;
-		pinctrl-names = "default";
+		pinctrl-names = "rockchip,camera_default", "rockchip,camera_sleep";
 		pinctrl-0 = <&cam_clkout0>;
+		pinctrl-1 = <&cam_sleep>;
 		//reset pin control by hardware,used this pin switch to mipi input
 		//1->2LANE(LANE 0&1) FRONT camera, 0->4LANE REAR camera
 		reset-gpios = <&gpio4 17 GPIO_ACTIVE_LOW>;
@@ -911,6 +912,12 @@
 				<4 RK_PA7 RK_FUNC_1 &pcfg_pull_none>;
 		};
 
+		cam_sleep: cam-sleep {
+			rockchip,pins =
+				/* cam_sleep */
+				<4 RK_PA7 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
 		camera_rst: camera-rst {
 			rockchip,pins =
 				/* front camera reset */
-- 
2.35.3

